### DISCUSSION TITLE
Tool function execution demo

### DISCUSSION THREAD

**Discussion by philipp-meier** [awaescher/OllamaSharp#70] [Ideas]
I started using Ollama, because I learned about the tool calls and wanted to try them out locally and found your tool demo.
In my opinion, it would be nice, if the tool console demo could also simulate simple function executions, so that the whole tool call functionality can be demonstrated/tested.

What do you think about this? Would this fit into the tool console demo?
I already implemented a simple version locally with this [commit](https://github.com/awaescher/OllamaSharp/compare/main...philipp-meier:OllamaSharp:demos/tool-invocation).

If someone is interested, I could create a PR with the mentioned commit.
... and if the demo was meant to only demonstrate the tool call itself or someone else has a better solution, that's fine too of course.

Thanks!

**Comment by awaescher** [comment]
Would love to see it. I confess the current tool demo is not really sexy. I just wanted to dogfood my implementation. Feel free to pimp it 😃

**Reply by philipp-meier** [reply]
I like it though. It is easily understandable and just works 😃 I created PR #73, if you are interested in this version. If you want to have further changes, just let me know. Thanks!

**Reply by awaescher** [reply]
Just merged it, thanks a lot ☺️
